### PR TITLE
fix(tests): EIP-7620,EIP-3540: Failing EOF tests

### PR DESCRIPTION
**Pull request by marioevz**
## 🗒️ Description
Makes the following fixes:
- `test_code_section_header_body_mismatch`: Change expected exception from `EOFException.INVALID_SECTION_BODIES_SIZE` -> `EOFException.INVALID_TYPE_SECTION_SIZE`
- `test_legacy_initcode_invalid_eof_v1_contract::trailing_bytes_after_data_section`: Correct output (`NON_RETURNING_SECTION`) in code section
- `test_eofcreate_deploy_sizes::64k-1` skip test due to oversized container in the pre-alloc.

## 🔗 Related Issues
None

## ✅ Checklist
- [x] All: Set appropriate labels for the changes.
- [x] All: Considered squashing commits to improve commit history.
- [ ] All: Added an entry to [CHANGELOG.md](/ethereum/execution-spec-tests/blob/main/docs/CHANGELOG.md).
- [x] All: Considered updating the online docs in the [./docs/](/ethereum/execution-spec-tests/blob/main/docs/) directory.
- [x] Tests: All converted JSON/YML tests from [ethereum/tests](/ethereum/tests) have been added to [converted-ethereum-tests.txt](/ethereum/execution-spec-tests/blob/main/converted-ethereum-tests.txt).
- [x] Tests: Included the type and version of evm t8n tool used to locally execute test cases:  e.g., ref with commit hash or geth 1.13.1-stable-3f40e65.
- [x] Tests: Ran `mkdocs serve` locally and verified the auto-generated docs for new tests in the [Test Case Reference](https://ethereum.github.io/execution-spec-tests/main/tests/) are correctly formatted.

### CODE REVIEW

**Comment by marioevz** [comment]
PTAL @shemnon @chfast @gumb0

File: tests/prague/eip7692_eof_v1/eip7620_eof_create/test_eofcreate_failures.py [hunk 1]
```diff
@@ -168,7 +168,11 @@ def test_initcode_aborts(
         pytest.param(MAX_BYTECODE_SIZE + 1, id="overmax"),
         pytest.param(MAX_INITCODE_SIZE - factory_size, id="initcodemax"),
         pytest.param(MAX_INITCODE_SIZE - factory_size + 1, id="initcodeovermax"),
-        pytest.param(0xFFFF - factory_size, id="64k-1"),
+        pytest.param(
+            0xFFFF - factory_size,
+            id="64k-1",
+            marks=pytest.mark.skip("Oversized container in pre-alloc"),
```
Comment (shemnon) [inline tests/prague/eip7692_eof_v1/eip7620_eof_create/test_eofcreate_failures.py:174]: Should the prior line get this skip mark as well?  There is a pending spec change to set max container sizes and initcodemax is a front runner.

**Review by shemnon — APPROVED** [review]

File: tests/prague/eip7692_eof_v1/eip7620_eof_create/test_eofcreate_failures.py (hunk 1 above)
Comment (marioevz) [reply tests/prague/eip7692_eof_v1/eip7620_eof_create/test_eofcreate_failures.py:174]: Yes, and actually, shouldn't be using a container creating tx a better place to put these larger container tests? Maybe have both cases ideally.

Comment (marioevz) [reply tests/prague/eip7692_eof_v1/eip7620_eof_create/test_eofcreate_failures.py:178]: Added in the last commit, also added the function signature for the same test but using a create tx.

**Comment by pdobacz** [comment]
@marioevz Just FYI, due to issues around

> test_eofcreate_deploy_sizes::64k-1 skip test due to oversized container in the pre-alloc.

we're considering putting a hard limit on the container size _during eof validation_, c.f. https://github.com/ipsilon/eof/pull/125 . This would define the maximum expected size to handle in tests.

**Pull request merged**